test/Microsoft.AspNetCore.Mvc.IntegrationTests/MutableObjectModelBinderIntegrationTest.cs

```diff
@@ -2126,6 +2126,51 @@ private class Address13
             Assert.Equal(ModelValidationState.Valid, entry.ValidationState);
         }
 
+        private class Product
+        {
+            public int ProductId { get; set; }
+
+            public string Name { get; }
+
+            public IList<string> Aliases { get; }
+        }
+
+        [Theory]
+        [InlineData("?parameter.ProductId=10")]
+        [InlineData("?parameter.ProductId=10&parameter.Name=Camera")]
+        [InlineData("?parameter.ProductId=10&parameter.Name=Camera&parameter.Aliases[0]=Camera1")]
+        public async Task MutableObjectModelBinder_BindsSettableProperties(string queryString)
```

Comment:
All references to the `MutableObjectModelBinder` are out-of-date. No need to clean everything up in this PR but stick with the current name `ComplexTypeModelBinder` in your additions. Same for comment below.